network/p2p/p2p.go

```diff
@@ -108,18 +110,37 @@ func MakeHost(cfg config.Local, datadir string, pstore *pstore.PeerStore) (host.
 	ua := fmt.Sprintf("algod/%d.%d (%s; commit=%s; %d) %s(%s)", version.Major, version.Minor, version.Channel, version.CommitHash, version.BuildNumber, runtime.GOOS, runtime.GOARCH)
 
 	var listenAddr string
+	var needAddressFilter bool
 	if cfg.NetAddress != "" {
 		if parsedListenAddr, perr := netAddressToListenAddress(cfg.NetAddress); perr == nil {
 			listenAddr = parsedListenAddr
+
+			// check if the listen address is a specific address or a "all interfaces" address (0.0.0.0 or ::)
+			// in this case enable the address filter.
+			// this also means the address filter is not enabled for NetAddress set to
+			// a specific address including loopback and private addresses.
+			if manet.IsIPUnspecified(multiaddr.StringCast(listenAddr)) {
+				needAddressFilter = true
+			}
+		} else {
+			logging.Base().Warnf("failed to parse NetAddress %s: %v", cfg.NetAddress, perr)
 		}
 	} else {
-		// don't listen if NetAddress is not set.
+		logging.Base().Debug("p2p NetAddress is not set, not listening")
 		listenAddr = ""
 	}
 
 	var enableMetrics = func(cfg *libp2p.Config) error { cfg.DisableMetrics = false; return nil }
 	metrics.DefaultRegistry().Register(&metrics.PrometheusDefaultMetrics)
 
+	// if we are listening on a loopback address, most likely we are running tests or a goal network,
```

Comment:
This seems like a weird spot for this comment, as the logic setting `needAddressFilter` is above.

Reply:
this left from previous iteration, I'll remove